### PR TITLE
Ensure profile gets reported on application shutdown

### PR DESCRIPTION
**Pull request by ivoanjo**
The `#work_pending?` callback is used to signal to `Datadog::Workers::IntervalLoop` when there is more work to be done.

During shutdown, when `#stop` is called on the `Scheduler`, this flag is used to determine if there is still work to do before `Scheduler` stops.

Thus, by correctly setting it to reflect the state of the recorder, we ensure that on application shutdown, any pending events are still reported; previously they would just be dropped.

Finally, the `#work_pending?` also needs to take into account the `#skip_next_flush?` so that we still preserve the behavior introduced in #1462 where we want the first `#flush_event` to always execute, but do nothing.

(Note: Current base branch is `fix/avoid-flush-after-start` since this builds off the change in #1462)

### CODE REVIEW

**Review by marcotc — DISMISSED** [review]

**Comment by ivoanjo** [comment]
(I'm waiting for #1462 to be merged before rebasing and merging this one)

**Comment by ivoanjo** [comment]
Now that #1479 was merged, I've re-done this PR to match it, and changed the target branch to master.

@marcotc can you take another look? :)

**Review by marcotc — APPROVED** [review]

**Pull request merged**
